test/test_allocator.cpp

```diff
@@ -110,3 +112,29 @@ TEST(test_allocator, realloc_failing_allocators) {
   auto failing_allocator = get_failing_allocator();
   EXPECT_EQ(nullptr, rcutils_reallocf(allocated_memory, 1024, &failing_allocator));
 }
+
+TEST(test_allocator, default_allocator_maybe_fail) {
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+
+  // Check allocate
+  rcutils_fault_injection_set_count(0);
```

Comment:
Could this be: 
```suggestion
  rcutils_fault_injection_set_count(RCUTILS_FAULT_INJECTION_FAIL_NOW);
```

Reply:
They are equivalent, so if you think the suggestion works better then I can go with it.  I think I just went with 0 in the odd event that this test has to be updated and someone needs to add a `set_count(1)` (or change the current case to `set_count(1)`) to handle another case.

Reply:
In that case, it will be better to have a complete `RCUTILS_FAULT_INJECTION_TEST` test, right?

I thought it may look better with the macro, so it is clear that 0 means to fail now. Feel free to leave as you think is more convenient.

Reply:
> In that case, it will be better to have a complete RCUTILS_FAULT_INJECTION_TEST test, right?

`RCUTILS_FAULT_INJECTION_TEST` always runs one more time than necessary in order to check the nominal case. If I used `RCUTILS_FAULT_INJECTION_TEST`, then the unit test would just have to get more complicated I think. I updated it to use `FAIL_NOW` as I think if code changes and this unit test fails, then updating this unit test should require a bit more review.

That all being said, I doubt default allocator will change often so this unit test should be pretty stable.